pkg/liquidity-mining/contracts/fee-distribution/FeeDistributor.sol

```diff
@@ -0,0 +1,501 @@
+        uint256 weekCursor = _userTimeCursor[user];
+        if (weekCursor == 0) {
+            // First checkpoint for user so need to do the initial binary search
+            userEpoch = _findTimestampUserEpoch(user, _startTime, maxUserEpoch);
+        } else {
+            if (weekCursor == _roundDownTimestamp(block.timestamp)) {
+                // User has checkpointed this week already so perform early return
+                return;
+            }
+            // Otherwise use the value saved from last time
+            userEpoch = _userLastEpochCheckpointed[user];
+        }
+
+        // Epoch 0 is always empty so bump onto the next one so that we start on a valid epoch.
+        if (userEpoch == 0) {
+            userEpoch = 1;
+        }
+
+        IVotingEscrow.Point memory userPoint = _votingEscrow.user_point_history(user, userEpoch);
+
+        // If this is the first checkpoint for the user, calculate the first week they're eligible for.
+        // i.e. the timestamp of the first Thursday after they locked.
+        if (weekCursor == 0) {
+            weekCursor = _roundUpTimestamp(userPoint.ts);
+        }
+
+        // Sanity check - can't claim fees from before fee distribution started.
+        if (weekCursor < _startTime) {
+            weekCursor = _startTime;
+        }
+
+        IVotingEscrow.Point memory oldUserPoint;
+        for (uint256 i = 0; i < 50; ++i) {
+            // Break if we're trying to cache the user's balance at a timestamp in the future
+            if (weekCursor > block.timestamp) {
+                break;
+            }
+
+            if (weekCursor >= userPoint.ts && userEpoch <= maxUserEpoch) {
+                // The week being considered is contained in an epoch after the user epoch described by `oldUserPoint`.
+                // We then shift `userPoint` into `oldUserPoint` and query the Point for the next user epoch.
+                // We do this in order to step though epochs until we find the last epoch starting before `weekCursor`.
+                userEpoch += 1;
+                oldUserPoint = userPoint;
+                if (userEpoch > maxUserEpoch) {
+                    userPoint = IVotingEscrow.Point(0, 0, 0, 0);
+                } else {
+                    userPoint = _votingEscrow.user_point_history(user, userEpoch);
+                }
+            } else {
+                // The week being considered lies inside the user epoch described by `oldUserPoint`
+                // we can then use it to calculate the user's balance at the beginning of the week.
+
+                int128 dt = int128(weekCursor - oldUserPoint.ts);
+                uint256 userBalance = oldUserPoint.bias > oldUserPoint.slope * dt
+                    ? uint256(oldUserPoint.bias - oldUserPoint.slope * dt)
+                    : 0;
+
+                // User's lock has expired and they haven't relocked yet.
+                if (userBalance == 0 && userEpoch > maxUserEpoch) break;
```

Comment:
if (userBalance == 0 && userEpoch > maxUserEpoch) {
                    weekCursor = _roundUpTimestamp(block.timestamp);
                    break;
                }

Reply:
Suggestion from Nico, we can fast forward the week cursor to now as the clearly hasn't received any rewards in the intervening weeks.